compliance/collection/nodeinventorizer/caching_inventorizer_test.go

```diff
@@ -0,0 +1,128 @@
+package nodeinventorizer
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stretchr/testify/suite"
+)
+
+type mockSleeper struct {
+	receivedDuration time.Duration
+	callCount        int
+}
+
+func (ms *mockSleeper) mockWaitCallback(d time.Duration) {
+	ms.receivedDuration = d
+	ms.callCount++
+}
+
+type mockScanner struct{}
+
+func (m mockScanner) Scan(nodeName string) (*storage.NodeInventory, error) {
+	return &storage.NodeInventory{NodeName: nodeName}, nil
+}
+
+type TestComplianceCachingSuite struct {
+	suite.Suite
+}
+
+func TestComplianceCaching(t *testing.T) {
+	suite.Run(t, new(TestComplianceCachingSuite))
+}
+
+// run before each test
+// func (s *TestComplianceCachingSuite) SetupTest() {
+//	s.sleeper = mockSleeper{callCount: 0}
+//	s.cs = *NewCachingScanner(
+//		fmt.Sprintf("%s/inventory-cache", s.T().TempDir()),
+//		s.sleeper.mockWaitCallback,
+//	)
+//}
+
+func (s *TestComplianceCachingSuite) TestMin() {
+	initial := 2 * time.Second
+	maxBackoff := 10 * time.Second
+
+	actual := min(initial, maxBackoff)
+
+	// On successful test, the duration must not be overwritten
+	s.Equal(initial, actual)
+}
+
+func (s *TestComplianceCachingSuite) TestMinMaxBackoff() {
+	initial, _ := time.ParseDuration("100h")
+	maxBackoff, _ := time.ParseDuration("10s")
+
+	actual := min(initial, maxBackoff)
+
+	s.Equal(maxBackoff, actual)
+}
+
+func (s *TestComplianceCachingSuite) TestCalcNextBackoff() {
+	initial, _ := time.ParseDuration("2s")
+	cache := initial
+	maxBackoff, _ := time.ParseDuration("10s")
+	cs := *NewCachingScanner(mockScanner{}, "", cache, initial, maxBackoff, func(time.Duration) {})
+	expectedBackoff := initial * backoffMultiplier
+
+	newBackoff := cs.calcNextBackoff(initial)
+
+	s.Equal(expectedBackoff, newBackoff)
+}
+
+func (s *TestComplianceCachingSuite) TestCalcNextBackoffUpperBoundary() {
+	initial, _ := time.ParseDuration("8s")
+	cache := initial
+	maxBackoff, _ := time.ParseDuration("10s")
+	cs := *NewCachingScanner(mockScanner{}, "", cache, initial, maxBackoff, func(time.Duration) {})
+
+	newBackoff := cs.calcNextBackoff(initial)
+
+	s.Equal(maxBackoff, newBackoff)
+}
```

Comment:
I see these tests are very similar. Could we merge them into a single table-driven test?

Reply:
Well, I see a lot of code duplication here. These tests have 2 inputs: initial, maxBackoff and one expected value returned by `calcNextBackoff`. It looks for me like a perfect candidate for a table-based test. 

(code tested)

```go
func (s *TestComplianceCachingSuite) TestCalcNextBackoffTable() {
	cases := map[string]struct {
		initial         time.Duration
		maxBackoff      time.Duration
		expectedBackoff time.Duration
	}{
		"next backoff should not hit the limit of 30s": {
			initial:         10 * time.Second,
			maxBackoff:      30 * time.Second,
			expectedBackoff: 15 * time.Second,
		},
		"next backoff should not be higher than the limit of 10s": {
			initial:         8 * time.Second,
			maxBackoff:      10 * time.Second,
			expectedBackoff: 10 * time.Second,
		},
	}
	for name, c := range cases {
		s.Run(name, func() {
			cs := NewCachingScanner(mockScanner{}, "", c.initial, c.initial, c.maxBackoff, func(time.Duration) {})
			s.Equal(c.expectedBackoff, cs.calcNextBackoff(c.initial))
		})
	}
}
```


No need to be the messenger if Misha requested the code to look differently. Let me discuss it with him. @msugakov would you be okay with using my table-driven test instead the two tests:  `TestCalcNextBackoff`, `TestCalcNextBackoffUpperBoundary`?

Reply:
I like the way this test is structured, I added this instead 😄